### PR TITLE
Add navigation via info.xml (#26785)

### PR DESCRIPTION
**Pull request by LukasReschke**
* Read navigation information from info.xml

* Load files navigation elements from info.xml

* Add comment about ignoring the exception

From https://github.com/owncloud/core/pull/26785

### CODE REVIEW

File: apps/files/appinfo/info.xml [hunk 1]
```diff
@@ -53,4 +53,10 @@
 		<command>OCA\Files\Command\DeleteOrphanedFiles</command>
 		<command>OCA\Files\Command\TransferOwnership</command>
 	</commands>
+
+	<navigation>
+		<route>files.view.index</route>
+		<order>0</order>
+	</navigation>
```
Comment (LukasReschke) [inline apps/files/appinfo/info.xml]: Once merged requires to be added to https://github.com/nextcloud/appstore/blob/master/nextcloudappstore/core/api/v1/release/info.xsd as well.

Comment (nickvergessen) [inline apps/files/appinfo/info.xml]: What I dislike about this is it drops the name tag by default, uses firstCaseUpper(appname). I think it's a trap for apps which then will accidently not have this translated anymore.

File: apps/files/img/app.svg [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<svg xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns="http://www.w3.org/2000/svg" height="32" width="32" version="1.1" xmlns:cc="http://creativecommons.org/ns#" xmlns:dc="http://purl.org/dc/elements/1.1/">
```
Comment (nickvergessen) [inline apps/files/img/app.svg]: no viewbox no minification

File: apps/files/appinfo/info.xml (hunk 1 above)
Comment (nickvergessen) [inline apps/files/appinfo/info.xml]: Also this looks like an app can only have one? Should we wrap it in a "navigations" so we can have muliple?
Or should those apps just use the programmatic way then instead?

**Review by nickvergessen — CHANGES_REQUESTED** [review]

File: apps/files/appinfo/info.xml (hunk 1 above)
Comment (LukasReschke) [inline apps/files/appinfo/info.xml]: > What I dislike about this is it drops the name tag by default, uses firstCaseUpper(appname). I think it's a trap for apps which then will accidently not have this translated anymore.

We should probably work on getting our CI scripts running then in a way that it supports the `lang=` tag as seen at http://nextcloudappstore.readthedocs.io/en/latest/developer.html#info-xml

File: apps/files/img/app.svg (hunk 2 above)
Comment (MorrisJobke) [inline apps/files/img/app.svg]: Done

File: apps/files/appinfo/info.xml (hunk 1 above)
Comment (nickvergessen) [inline apps/files/appinfo/info.xml]: Can also adjust the travis thing to translate `<navigation><name>` when its provided?

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍 

Regarding multiple entries: I would keep the info.xml with one entry. For multiple entries the app needs to use the normal OCP API.

File: apps/files/appinfo/info.xml (hunk 1 above)
Comment (nickvergessen) [inline apps/files/appinfo/info.xml]: @MorrisJobke can we add this simply? Extracting `<navigation><name>` automatically to travis?

Comment (MorrisJobke) [inline apps/files/appinfo/info.xml]: > @MorrisJobke can we add this simply? Extracting <navigation><name> automatically to travis?

Create me a ticket - I will look into extending it 😉

**Review by nickvergessen — APPROVED** [review]
I made the name non-optional.
I'm fine with this now and will create the translation issue.

**Comment by MorrisJobke** [comment]
:+1: (ignore CI -.- there seems to be an issue with the autoloader check - I guess they released a new version between Job #1 and #2 - so one failed and one succeeded)

I will merge this.

**Pull request merged**